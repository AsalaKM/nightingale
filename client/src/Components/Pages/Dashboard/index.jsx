import React, { Component } from "react";
import Chart from "../../Common/Chart";
import uuid from "uuid";
import axios from "axios";
import {
  TopDiv,
  Name,
  Welcome,
  Message,
  DailyDiv,
  Daily,
  Dailyparagraph
} from "./index.style";

class Dashboard extends Component {
  state = {
    message: "",
    dailyMood: [],
    statusMood: null
  };

  componentDidMount = () => {
    axios
      .get("/api/user/dashboard")
      .then(res => {
        const { data } = res;
        if (
          data[0].moodsBystatus.length === 0 &&
          data[0].moodsByDays.length === 0
        ) {
          const message =
            "hey! there is no information about your mood! Start chatting to get it";
          this.setState({ message, dailyMood: [], statusMood: null });
        } else {
          const { moodsBystatus, moodsByDays } = data[0];
          // Put the mood by status in chart object to show
          const statusMood = moodsBystatus.map(row => ({
            decription: row.moodEmoji,
            percentage: row.theAverage.toFixed(1),
            mood: row.moodDescription,
            count: row.count
          }));
          const { dailyMood } = this.state;
          // Get the mood by days
          moodsByDays.map(row => dailyMood.push([row.moodEmoji, row._id]));
          // Set new data in the State
          this.setState({ statusMood, dailyMood });
        }
      })
      .catch(() => {
        const msg = " Sorry, There is an error!!";
        this.setState({ message: msg, conversations: [] });
      });
  };

  render() {
    const { message, statusMood, dailyMood } = this.state;
    return (
      <>
        <TopDiv>
          <Name>Hi {this.props.name} !</Name>
          <Welcome>It’s great to see you again</Welcome>
          {message && <Message>{message}</Message>}
          {statusMood && dailyMood && (
            <Message>
              Here’s a breakdown of how you’ve felt before each of our
              conversations over the last 30 days:
            </Message>
          )}
        </TopDiv>

        {statusMood && <Chart sections={statusMood} width={200} />}
        {dailyMood && (
          <DailyDiv>
            {dailyMood.map(day => (
              <Daily key={uuid()}>
                {day.map(value => (
                  <Dailyparagraph key={uuid()}>{value}</Dailyparagraph>
                ))}
              </Daily>
            ))}
          </DailyDiv>
        )}
      </>
    );
  }
}
export default Dashboard;
