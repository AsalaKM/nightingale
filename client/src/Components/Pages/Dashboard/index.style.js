import styled from "styled-components";
export const View = styled.div`
margin-bottom: 5rem;
  img {
    margin-bottom: 15px;
    margin-right:20px
    position: fixed;
    bottom: 0;
    cursor:pointer;
    right: 0;
  }
`;

export const TopDiv = styled.div`
  width: 95%;
  margin: 5px auto 15px auto;
  padding-lift: 20px;
`;
export const Name = styled.h1`
  margin-bottom: 12px;
`;
export const Welcome = styled.h2`
  margin-bottom: 12px;
`;
export const Message = styled.p`
  margin-bottom: 15px;
`;

export const DailyDiv = styled.div`
  display: flex;
  justify-content: space-around;
  width: 50%;
  margin: 0 auto;
  margin-top: 3rem;
  border-top: #7874fd solid 1px;
  padding-top: 20px;
  @media (max-width: 768px) {
    width: 95%;
  }
`;
export const Daily = styled.h5`
  @media (max-width: 768px) {
    font-size: 1rem;
  }
`;
export const Dailyparagraph = styled.p`
  text-align: center;
  animation: up 0.5s ease-in 0.5s;
  animation-fill-mode: both;
  :nth-child(1) {
    font-size: 1.5rem;
  }
  @keyframes up {
    0% {
      visibility: visible;
      opacity: 0;
      transform: translateY(1rem);
    }
    70% {
      transform: translateY(-0.1rem);
    }
    100% {
      opacity: 1;
      transform: translate(0);
    }
  }
  @media (max-width: 768px) {
    font-size: 0.5rem;
    :nth-child(1) {
      font-size: 1rem;
    }
  }
`;
