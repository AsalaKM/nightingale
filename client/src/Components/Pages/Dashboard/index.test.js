import React from "react";
import renderer from "react-test-renderer";
import Dashboard from "./index.jsx";
import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

// set enzyme configuration
configure({ adapter: new Adapter() });

describe("Dashboard", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Dashboard />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const wrapper = mount(<Dashboard />);
    const moodsBystatus = [
      {
        theAverage: 50,
        id: 1,
        count: 1,
        moodDescription: "terrible",
        moodEmoji: "ðŸ˜¡"
      },
      {
        theAverage: 50,
        id: 2,
        count: 1,
        moodDescription: "not great",
        moodEmoji: "ðŸ˜«"
      }
    ];
    const moodsByDays = [
      {
        _id: "Friday",
        moodEmoji: "ðŸ˜”"
      },
      {
        _id: "Thursday",
        moodEmoji: "ðŸ˜«"
      }
    ];

    const statusMood = moodsBystatus.map(row => ({
      decription: row.moodEmoji,
      percentage: row.theAverage.toFixed(1),
      mood: row.moodDescription,
      count: row.count
    }));
    const dailyMood = [];
    moodsByDays.map(row => dailyMood.push([row.moodEmoji, row._id]));

    // Save the data in the state
    wrapper.setState({ dailyMood, statusMood });
  });
});
