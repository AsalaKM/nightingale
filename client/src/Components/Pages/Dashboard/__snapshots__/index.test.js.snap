// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders correctly 1`] = `
<div
  className="sc-gzVnrw EFEtm"
>
  <div
    className="sc-htoDjs gmfqSo"
  >
    <h1
      className="sc-dnqmqq hTpVIo"
    >
      Hi 
       !
    </h1>
    <h2
      className="sc-iwsKbI jpysfj"
    >
      Itâ€™s great to see you again
    </h2>
    
  </div>
  <div
    className="sc-gqjmRU iJPaHM"
  >
    <p>
      Here's a breakdown of your average mood per day:
    </p>
    <div />
  </div>
</div>
`;
